src/jit/hwintrinsicxarch.cpp

```diff
@@ -1055,35 +1055,51 @@ GenTree* Compiler::impSSE42Intrinsic(NamedIntrinsic        intrinsic,
     return retNode;
 }
 
-GenTree* Compiler::impAVXIntrinsic(NamedIntrinsic        intrinsic,
-                                   CORINFO_METHOD_HANDLE method,
-                                   CORINFO_SIG_INFO*     sig,
-                                   bool                  mustExpand)
+GenTree* Compiler::impAvxOrAvx2Intrinsic(NamedIntrinsic        intrinsic,
+                                         CORINFO_METHOD_HANDLE method,
+                                         CORINFO_SIG_INFO*     sig,
+                                         bool                  mustExpand)
 {
     GenTree*  retNode  = nullptr;
     GenTree*  op1      = nullptr;
     GenTree*  op2      = nullptr;
     var_types baseType = TYP_UNKNOWN;
-    switch (intrinsic)
-    {
-        default:
-            JITDUMP("Not implemented hardware intrinsic");
-            break;
-    }
-    return retNode;
-}
+    int       simdSize = simdSizeOfHWIntrinsic(intrinsic, sig);
 
-GenTree* Compiler::impAVX2Intrinsic(NamedIntrinsic        intrinsic,
-                                    CORINFO_METHOD_HANDLE method,
-                                    CORINFO_SIG_INFO*     sig,
-                                    bool                  mustExpand)
-{
-    GenTree*  retNode  = nullptr;
-    GenTree*  op1      = nullptr;
-    GenTree*  op2      = nullptr;
-    var_types baseType = TYP_UNKNOWN;
     switch (intrinsic)
     {
+        case NI_AVX_ExtractVector128:
+        case NI_AVX2_ExtractVector128:
+        {
+            if (impStackTop().val->IsCnsIntOrI() || mustExpand)
```

Comment:
Why isn't this being handled above with the rest of the constant handling?

Reply:
Thanks, fixed.